### DISCUSSION TITLE
Example of routing with parameters

### DISCUSSION THREAD

**Discussion by nlfiedler** [leptos-rs/leptos#3111] [General]
I am looking for an example of routing with parameters. The `static_routing` example uses a lot of private modules inside `leptos`, so that does not help (would that be considered a bug since I cannot use the example?). The book briefly touches on the topic, but I don't see any example of writing a component that receives the URL parameter (e.g. `UserProfile`).

Is there an example that would shed light on how a component can receive URL parameters?

Thank you.

**Comment by gbj** [comment]
I'm not sure what you mean about the static routing example. The `router` example uses route parameters. If this isn't what you mean, could you please clarify?

**Comment by nlfiedler** [comment]
I'm mixing up code lines so that isn't helping, but the examples in the `main` branch are using modules that are private in the 0.6 release that I am using. What I was looking for was an example of a component, in the book, that showed how to get the parameters. Pieces of it are there, if you look closely, in the params and queries chapter, or if you happen to expand the sandbox example at the bottom of the page. The next best place to look is the `examples` directory, but I was looking at the wrong code line and that led me here.

Thank you for your help.

**Reply by gbj** [reply]
To avoid additional confusion, note also that the main branch right now is 0.7 (see README). To find 0.6 examples look at the `leptos_0.6` branch.
